topo_processor/metadata/metadata_loaders/metadata_loader_tiff.py

```diff
@@ -23,18 +25,40 @@ def is_applicable(self, asset: Asset) -> bool:
         return is_tiff(asset.source_path)
 
     def load_metadata(self, asset: Asset) -> None:
-        asset.item.add_extension(stac.StacExtensions.projection.value)
 
         fs = get_fs(asset.source_path)
         with fs.open(asset.source_path) as f:
             with rasterio.open(f) as tiff:
-                if tiff.crs:
-                    if not tiff.crs.is_epsg_code:
-                        raise Exception("The code is not a valid EPSG code.")
-                    crs = tiff.crs.to_epsg()
-                else:
-                    crs = None
-                asset.item.properties.update({"proj:epsg": crs})
+                self.add_epsg(tiff, asset)
+                self.add_bands(tiff, asset)
 
     def load_all_metadata(self, metadata_file: str) -> None:
         pass
+
+    def add_epsg(self, tiff, asset):
+        if tiff.crs:
+            if not tiff.crs.is_epsg_code:
+                raise Exception("The code is not a valid EPSG code.")
+            crs = tiff.crs.to_epsg()
+        else:
+            crs = None
+        asset.item.properties["proj:epsg"] = crs
+        asset.item.add_extension(stac.StacExtensions.projection.value)
+
+    def add_bands(self, tiff, asset):
+        if ColorInterp.gray in tiff.colorinterp and len(tiff.colorinterp) == 1:
+            asset.properties["eo:bands"] = [{"name": ColorInterp.gray.name, "common_name": "pan"}]
+            asset.item.add_extension(stac.StacExtensions.eo.value)
```

Comment:
The eo extension should always be added to stac_extensions or validation will fail.